### PR TITLE
Controllers

### PR DESCRIPTION
**Pull request by AndrePu**

### CODE REVIEW

File: Cooper/ClientApp/angular.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "version": 1,
   "newProjectRoot": "projects",
   "projects": {
-    "Cooper": {
+    "my-first-project": {
```
Comment (Hunroll) [inline Cooper/ClientApp/angular.json]: Leave "Cooper"

File: Cooper/Controllers/GameReviewController.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Cooper.Models;
+
+// For more information on enabling Web API for empty projects, visit https://go.microsoft.com/fwlink/?LinkID=397860
+
+namespace Cooper.Controllers
+{
+    [Route("api/[controller]")]
```
Comment (Hunroll) [inline Cooper/Controllers/GameReviewController.cs]: Different format of Controller's route. Games have "api/games", here you have "api/[controller]" choose a single way of providing a route

File: Cooper/Controllers/GameReviewController.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+        // POST api/<controller>
+        [HttpPost]
+        public IActionResult Post([FromBody]GameReview gameReview)
+        {
+            return Ok(new GameReview());
+        }
+
+        // PUT api/<controller>/5
+        [HttpPut("{id}")]
```
Comment (Hunroll) [inline Cooper/Controllers/GameReviewController.cs]: Method is not needed. You should use the previous one (POST@api/review) and post the json with id

File: Cooper/Controllers/SampleDataController.cs [hunk 4]
```diff
@@ -7,38 +7,72 @@
 namespace Cooper.Controllers
```
Comment (Hunroll) [inline Cooper/Controllers/SampleDataController.cs]: drop this file, it's not needed. If you need it for testing - save it locally.

File: Cooper/DAO/GameDAO.cs [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+{
+    public class GameDAO : IDAO<GameDb>
+    {
+
+        private CRUD crud;
+        Logger logger;
+
+        private string table;
+        private string idColumn;
+        private List<string> attributes;
```
Comment (Hunroll) [inline Cooper/DAO/GameDAO.cs]: "Set" is a better structure for it. You can read about most common data structures and its usage area.

File: Cooper/DAO/GameDAO.cs [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+                }
+            }
+
+            return game;
+        }
+        
+        ///<summary>
+        ///Maps properties from GameDb object to EntityORM object
+        ///</summary>
+        private EntityORM MapEntity(GameDb game)
```
Comment (Hunroll) [inline Cooper/DAO/GameDAO.cs]: SOLID, letter S states for single responsibility, so one class has a single job. Move mappers to another class Mapper which will have all methods like 
public EntityOrm ToOrm(GameDb game), public EntityOrm ToOrm(ChatDb chat), ...
and reverse methods:
public GameDb ToGame(EntityOrm entity), public ChatDb ToChat(EntityOrm chat), ....

File: Cooper/ORM/EntityORM.cs [hunk 7]
```diff
@@ -0,0 +1,16 @@
+using System;
+using System.Collections.Generic;
+
+namespace Cooper.ORM
+{
+    public class EntityORM
+    {
+        public List<AttributeValue> attributeValue {get;set;}
```
Comment (Hunroll) [inline Cooper/ORM/EntityORM.cs]: make it a dictionary and you won't have the class AttributeValue. Dictionary is a perfect structure for it

File: Cooper/ClientApp/angular.json (hunk 1 above)
Comment (AndrePu) [reply Cooper/ClientApp/angular.json]: Client group work

**Pull request merged**